### ISSUE TITLE
Feature Request

### ISSUE THREAD

**Issue by JH1979** [gagax1234/OpenFunscripter#10]

**Comment by JH1979** [comment]
Hello,First a big Thank You for that cool Software!

I have a Feature Request, no Clue if it is even possible, but I'll try to explain.
If scripting T-Code and Viewing the T-Code, do You think it would be possible to create points and manipulate the Points with the Indicator-Slider (in the Output View)? this would be a cool Feature, because You could watch the 3D Simulator while manipulating, would be very usefull for Roll and Pitch-Axis on the OSR2.

**Comment by gagax1234** [comment]
So here's what I got from that.
You want to script using the T-Code sliders.
Now while technically possible the structure of the code doesn't allow and that's a good thing and I'll keep it that way.
The T-Code player is for playing T-Code and for nothing else.

What I can do is add that sort of thing to the 3D simulator with sliders and maybe other things too.

**Comment by JH1979** [comment]
not exactly scripting, manipulating of Points would be enough, It doesn't have to be in the Tcode Window, I just thought because these Sliders/Bars where already there, to add it to the 3D Simulator would also be cool, and probably the Better way, to keep things clearly segmented.

**Comment by gagax1234** [comment]
Well I did something.
Let me know if this is what you had in mind. Personally I think it's very clunky.
In the 3d simulator I added a "Scripting" tab.
If you have your loaded scripts hooked up to the correct axis you'll have 3 sliders which you can drag see, live feedback on the simulator and if you release the mouse points get inserted/edited.

Protip: Loading the scripts in the following order: stroke, roll, pitch, twist makes it so it's always configured correctly.

https://ci.appveyor.com/api/buildjobs/65mbt0mg12temcv6/artifacts/bin%2FRelease%2FOFS-x64_Windows_master-6fa53c6e5a59ae81564a4d25bf66a9010bd8a13f.7z

**Comment by JH1979** [comment]
Awesome!

 

That's exactly was I was hoping for!

 

A BIG THANK YOU!

 

This will make the Whole Scripting-Process a lot easyer!

 

Greetings!

 

 

Gesendet: Montag, 08. März 2021 um 22:51 Uhr
Von: "gagax1234" <notifications@github.com>
An: "gagax1234/OpenFunscripter" <OpenFunscripter@noreply.github.com>
Cc: "JH1979" <jhainer@gmx.de>, "Author" <author@noreply.github.com>
Betreff: Re: [gagax1234/OpenFunscripter] Feature Request (#10)


 

Well I did something.
Let me know if this is what you had in mind. Personally I think it's very clunky.
In the 3d simulator I added a "Scripting" tab.
If you have your loaded scripts hooked up to the correct axis you'll have 3 sliders which you can drag see, live feedback on the simulator and if you release the mouse points get inserted/edited.

Protip: Loading the scripts in the following order: stroke, roll, pitch, twist makes it so it's always configured correctly.

https://ci.appveyor.com/api/buildjobs/65mbt0mg12temcv6/artifacts/bin%2FRelease%2FOFS-x64_Windows_master-6fa53c6e5a59ae81564a4d25bf66a9010bd8a13f.7z

**Comment by JH1979** [comment]
One more Question,

The Latest Implementation of the Sliders is already a great Improvement for speeding up the Workflow in Multi-Channel-Scripting, do You think it would be possible to use the Mouse-Wheel to set/manipulate the Values at Mouseover on the slider?
Because that way one could either use the Mouse or the Rotary-Controller on a Graphics-Tablet which would lead to further Time Savings. Thank You for all Your great Work ! ;-)

**Comment by gagax1234** [comment]
Yes added that.
I made some changes so that when you edit one axis, points also get added to the other axis which you aren't editing ( you'll notice what I mean when you try it).
I also added  a button to insert the current position & a variable to adjust how much scrolling affects the slider.
By default 1 scroll tick is +/-5 in the script.

The hardest part about this was getting this to work with the undo system but I think I did a good job. 😄 

https://ci.appveyor.com/api/buildjobs/7w2ixejeagsrrmny/artifacts/bin%2FRelease%2FOFS-x64_Windows_master-5e403006edc0df4d9328cb1d18e3db33241dee57.7z
